packages/spindle-ui/src/Pagination/hooks/useShowItem.test.ts

```diff
@@ -1,44 +1,601 @@
+      }),
+    );
+
+    expect(result.current).toEqual([1, 2, 3, 4, 6]);
+  });
+
+  it('use show item current 3 total 6', () => {
+    const current = 3;
+    const total = 6;
+
+    const { result } = renderHook(() =>
+      useShowItem({
+        current,
+        total,
+        totalThreshold: TOTAL_THRESHOLD,
+        maxShowItemSize: MAX_SHOW_ITEM_SIZE,
+      }),
+    );
+
+    expect(result.current).toEqual([1, 3, 4, 5, 6]);
+  });
+
+  it('use show item current 4 total 6', () => {
+    const current = 4;
+    const total = 6;
+
+    const { result } = renderHook(() =>
+      useShowItem({
+        current,
+        total,
+        totalThreshold: TOTAL_THRESHOLD,
+        maxShowItemSize: MAX_SHOW_ITEM_SIZE,
+      }),
+    );
+
+    expect(result.current).toEqual([1, 3, 4, 5, 6]);
+  });
+
+  it('use show item current 5 total 6', () => {
+    const current = 5;
+    const total = 6;
+
+    const { result } = renderHook(() =>
+      useShowItem({
+        current,
+        total,
+        totalThreshold: TOTAL_THRESHOLD,
+        maxShowItemSize: MAX_SHOW_ITEM_SIZE,
+      }),
+    );
+
+    expect(result.current).toEqual([1, 3, 4, 5, 6]);
+  });
+
+  it('use show item current 6 total 6', () => {
+    const current = 6;
+    const total = 6;
+
+    const { result } = renderHook(() =>
+      useShowItem({
+        current,
+        total,
+        totalThreshold: TOTAL_THRESHOLD,
+        maxShowItemSize: MAX_SHOW_ITEM_SIZE,
+      }),
+    );
+
+    expect(result.current).toEqual([1, 3, 4, 5, 6]);
+  });
+
+  it('use show item current 1 total 7', () => {
+    const current = 1;
+    const total = 7;
+
+    const { result } = renderHook(() =>
+      useShowItem({
+        current,
+        total,
+        totalThreshold: TOTAL_THRESHOLD,
+        maxShowItemSize: MAX_SHOW_ITEM_SIZE,
+      }),
+    );
+
+    expect(result.current).toEqual([1, 2, 3, 4, 7]);
+  });
+
+  it('use show item current 2 total 7', () => {
+    const current = 2;
+    const total = 7;
+
+    const { result } = renderHook(() =>
+      useShowItem({
+        current,
+        total,
+        totalThreshold: TOTAL_THRESHOLD,
+        maxShowItemSize: MAX_SHOW_ITEM_SIZE,
+      }),
+    );
+
+    expect(result.current).toEqual([1, 2, 3, 4, 7]);
+  });
+
+  it('use show item current 3 total 7', () => {
+    const current = 3;
+    const total = 7;
+
+    const { result } = renderHook(() =>
+      useShowItem({
+        current,
+        total,
+        totalThreshold: TOTAL_THRESHOLD,
+        maxShowItemSize: MAX_SHOW_ITEM_SIZE,
+      }),
+    );
+
+    expect(result.current).toEqual([1, 3, 4, 5, 7]);
+  });
+
+  it('use show item current 4 total 7', () => {
+    const current = 4;
+    const total = 7;
+
+    const { result } = renderHook(() =>
+      useShowItem({
+        current,
+        total,
+        totalThreshold: TOTAL_THRESHOLD,
+        maxShowItemSize: MAX_SHOW_ITEM_SIZE,
+      }),
+    );
+
+    expect(result.current).toEqual([1, 4, 5, 6, 7]);
```

Comment:
<img width="280" alt="image" src="https://user-images.githubusercontent.com/4308418/204219214-de68a3b8-72d7-4987-b8d6-95346395551c.png">


```suggestion
    expect(result.current).toEqual([1, 3, 4, 5, 7]);
```